### ISSUE TITLE
[Bug]: [import v2]When importing parquet files generated using the bulk write, there is an error `expect 'Float' type for field 'array_float', but got 'list' type`

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#31834]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240329-88d426f39-amd64
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
[2024-04-02 17:57:40 - INFO - ci_test]: collection schema: {'auto_id': True, 'description': '', 'fields': [{'name': 'uid', 'description': '', 'type': <DataType.INT64: 5>, 'is_primary': True, 'auto_id': True}, {'name': 'int_scalar', 'description': '', 'type': <DataType.INT64: 5>}, {'name': 'float_scalar', 'description': '', 'type': <DataType.FLOAT: 10>}, {'name': 'string_scalar', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 65535}}, {'name': 'json', 'description': '', 'type': <DataType.JSON: 23>}, {'name': 'array_int', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_capacity': 100}, 'element_type': <DataType.INT64: 5>}, {'name': 'array_float', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_capacity': 100}, 'element_type': <DataType.FLOAT: 10>}, {'name': 'array_string', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_length': 100, 'max_capacity': 100}, 'element_type': <DataType.VARCHAR: 21>}, {'name': 'array_bool', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_capacity': 100}, 'element_type': <DataType.BOOL: 1>}, {'name': 'vectors', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}], 'enable_dynamic_field': True} (test_bulk_insert.py:1168)
[2024-04-02 17:57:40 - INFO - local_bulk_writer]: Data path created: /Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer (local_bulk_writer.py:76)
[2024-04-02 17:57:40 - INFO - local_bulk_writer]: Data path created: /Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55 (local_bulk_writer.py:81)
[2024-04-02 17:57:40 - INFO - remote_bulk_writer]: Minio/S3 blob storage client successfully initialized (remote_bulk_writer.py:156)
[2024-04-02 17:57:40 - INFO - remote_bulk_writer]: Remote buffer writer initialized, target path: /bulk_data/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55 (remote_bulk_writer.py:122)
[2024-04-02 17:57:41 - INFO - local_bulk_writer]: Prepare to flush buffer, row_count: 1000, size: 586000 (local_bulk_writer.py:108)
[2024-04-02 17:57:41 - INFO - local_bulk_writer]: Flush thread begin, name: Thread-10 (_flush) (local_bulk_writer.py:115)
[2024-04-02 17:57:41 - INFO - local_bulk_writer]: Wait flush to finish (local_bulk_writer.py:119)
[2024-04-02 17:57:41 - INFO - bulk_buffer]: Successfully persist file /Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet, total size: 586000, row count: 1000, row group size: 57162 (buffer.py:245)
[2024-04-02 17:57:41 - INFO - remote_bulk_writer]: Prepare to upload '/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet' to 'bulk_data/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet' (remote_bulk_writer.py:232)
[2024-04-02 17:57:41 - INFO - remote_bulk_writer]: Target bucket: 'milvus-bucket' (remote_bulk_writer.py:234)
[2024-04-02 17:57:41 - INFO - remote_bulk_writer]: Upload file '/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet' to 'bulk_data/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet' (remote_bulk_writer.py:256)
[2024-04-02 17:57:41 - INFO - remote_bulk_writer]: Successfully upload files: ['/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet'] (remote_bulk_writer.py:302)
[2024-04-02 17:57:41 - INFO - local_bulk_writer]: Flush thread done, name: Thread-10 (_flush) (local_bulk_writer.py:141)
[2024-04-02 17:57:41 - INFO - local_bulk_writer]: Commit done with async=False (local_bulk_writer.py:123)
[2024-04-02 17:57:41 - INFO - local_bulk_writer]: Delete local directory '/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55' (local_bulk_writer.py:87)
[2024-04-02 17:57:41 - INFO - remote_bulk_writer]: Delete empty directory '/Applications/PyCharm CE.app/Contents/plugins/python-ce/helpers/pycharm/bulk_writer' (remote_bulk_writer.py:134)
[2024-04-02 17:57:41 - INFO - ci_test]: before bulk load, there are 1 working tasks (utility_wrapper.py:25)
[2024-04-02 17:57:41 - INFO - ci_test]: files to load: ['bulk_data/6a9bffc7-fd0d-4dd1-8777-533fe4e47c55/1.parquet'] (utility_wrapper.py:26)
[2024-04-02 17:57:42 - INFO - ci_test]: after bulk load, there are 1 working tasks (utility_wrapper.py:34)
[2024-04-02 17:57:42 - INFO - root]: bulk insert task ids:448766837459462570 (test_bulk_insert.py:1208)
[2024-04-02 17:57:42 - INFO - ci_test]: wait bulk load timeout is 300 (utility_wrapper.py:111)
[2024-04-02 17:57:42 - INFO - ci_test]: before waiting, there are 1 pending tasks (utility_wrapper.py:113)
[2024-04-02 17:57:46 - INFO - ci_test]: after waiting, there are 0 pending tasks (utility_wrapper.py:148)
[2024-04-02 17:57:46 - INFO - ci_test]: task state distribution: {'success': set(), 'failed': {448766837459462570}, 'in_progress': {448766837459462570}} (utility_wrapper.py:149)
[2024-04-02 17:57:46 - INFO - ci_test]: {448766837459462570: <Bulk insert state:
    - taskID          : 448766837459462570,
    - state           : Failed,
    - row_count       : 0,
    - infos           : {'failed_reason': "expect 'Float' type for field 'array_float', but got 'list' type: importing data failed", 'progress_percent': '0'},
    - id_ranges       : [],
    - create_ts       : 2024-04-02 17:57:41
>} (utility_wrapper.py:150)
[2024-04-02 17:57:46 - INFO - ci_test]: wait for bulk load tasks completed failed, cost time: 4.06863808631897 (utility_wrapper.py:155)
[2024-04-02 17:57:46 - INFO - ci_test]: bulk insert state:False in 5.186257362365723 with states:{448766837459462570: <Bulk insert state:
    - taskID          : 448766837459462570,
    - state           : Failed,
    - row_count       : 0,
    - infos           : {'failed_reason': "expect 'Float' type for field 'array_float', but got 'list' type: importing data failed", 'progress_percent': '0'},
    - id_ranges       : [],
    - create_ts       : 2024-04-02 17:57:41
>} (test_bulk_insert.py:1213)
```

i print out the datatype of array_float and it is `{'name': 'array_float', 'description': '', 'type': <DataType.ARRAY: 22>, 'params': {'max_capacity': 100}, 'element_type': <DataType.FLOAT: 10>}`

```
        self._connect()
        fields = [
            cf.gen_int64_field(name=df.pk_field, is_primary=True, auto_id=auto_id),
            cf.gen_int64_field(name=df.int_field),
            cf.gen_float_field(name=df.float_field),
            cf.gen_string_field(name=df.string_field),
            cf.gen_json_field(name=df.json_field),
            cf.gen_array_field(name=df.array_int_field, element_type=DataType.INT64),
            cf.gen_array_field(name=df.array_float_field, element_type=DataType.FLOAT),
            cf.gen_array_field(name=df.array_string_field, element_type=DataType.VARCHAR, max_length=100),
            cf.gen_array_field(name=df.array_bool_field, element_type=DataType.BOOL),
            cf.gen_float_vec_field(name=df.vec_field, dim=dim),
        ]
        c_name = cf.gen_unique_str("bulk_insert")
        schema = cf.gen_collection_schema(fields=fields, auto_id=auto_id, enable_dynamic_field=enable_dynamic_field)
        log.info(f"collection schema: {schema}")
        self.collection_wrap.init_collection(c_name, schema=schema)
        with RemoteBulkWriter(
            schema=schema,
            remote_path="bulk_data",
            connect_param=RemoteBulkWriter.ConnectParam(
                bucket_name=self.bucket_name,
                endpoint=self.minio_endpoint,
                access_key="minioadmin",
                secret_key="minioadmin",
            ),
            file_type=BulkFileType.PARQUET,
        ) as remote_writer:
            for i in range(entities):
                row = {
                    df.pk_field: i,
                    df.int_field: 1,
                    df.float_field: 1.0,
                    df.string_field: "string",
                    df.json_field: {"key": "value"},
                    df.array_int_field: [1, 2],
                    df.array_float_field: [1.0, 2.0],
                    df.array_string_field: ["string1", "string2"],
                    df.array_bool_field: [True, False],
                    df.vec_field: [random.random() for _ in range(dim)]
                }
                if auto_id:
                    row.pop(df.pk_field)
                if enable_dynamic_field:
                    row["name"] = fake.name()
                    row["address"] = fake.address()
                remote_writer.append_row(row)
            remote_writer.commit()
            batch_files = remote_writer.batch_files
        for files in batch_files:
            # import data
            t0 = time.time()
            task_id, _ = self.utility_wrap.do_bulk_insert(
                collection_name=c_name, files=files
            )
            logging.info(f"bulk insert task ids:{task_id}")
            success, states = self.utility_wrap.wait_for_bulk_insert_tasks_completed(
                task_ids=[task_id], timeout=300
            )
            tt = time.time() - t0
            log.info(f"bulk insert state:{success} in {tt} with states:{states}")
            assert success
```

the parquet file
[1.parquet.zip](https://github.com/milvus-io/milvus/files/14834908/1.parquet.zip)

### Milvus Log

[standalone.log](https://github.com/milvus-io/milvus/files/14834943/standalone.log)

**Comment by zhuwenxing** [comment]
see test case https://github.com/zhuwenxing/milvus/blob/95fef0967c54ea90e428c470071faac7b4035089/tests/python_client/testcases/test_bulk_insert.py#L1150

**Comment by xiaofan-luan** [comment]
good catch!

**Comment by bigsheeper** [comment]
/assign @zhuwenxing 
please help to verify

**Comment by zhuwenxing** [comment]
verified and fix with `pymilvus-2.4.1rc10`
